### ISSUE TITLE
CA1801 is flagged for methods used as event handlers

### ISSUE THREAD

**Issue by scottdsanders** [dotnet/roslyn-analyzers#3106]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.8 (Latest)

#### Diagnostic ID

[CA1801](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1801?view=vs-2019)

#### Repro steps

1. Create a default XAML Win10 app.
2. Add a Page_Loaded event handler to the Page derived class like this:

```
private void Page_Loaded(object sender, RoutedEventArgs e)
{
    // ... do stuff without referencing sender or e
}

```
#### Expected behavior
Event handlers with mandatory signature are not flagged for unused variables.

#### Actual behavior

Seems like same issue as [False CA1801 reports (remove unused parameter) for methods used as event handlers #1218](https://github.com/dotnet/roslyn-analyzers/issues/1218) but it is still being flagged for event handlers...

**Comment by mavasani** [comment]
Interesting, we do seem to be bailing out on these: https://github.com/dotnet/roslyn-analyzers/blob/6d265703c1ccd10176dc77e768412cf1422f9153/src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/ReviewUnusedParameters.cs#L149-L156

**Comment by mavasani** [comment]
Tagging @Evangelink

**Comment by Evangelink** [comment]
@mavasani We only consider the method as event handler if the 2nd argument inherits from `EventArgs` which isn't the case for UWP applications. This is the case in @scottdsanders example, [RoutedEventArgs](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.routedeventargs).

To handle UWP arguments we could either replace `method.Parameters[1].Type.Inherits(eventsArgSymbol)` by a list of allowed base event args or we could allow for any parameter whose name ends in `EventArgs`. Note that for the second case, in order to avoid regression, we should keep the inherits and add the name check.

**Comment by mavasani** [comment]
@Evangelink Lets start by just treating `RoutedEventArgs` specially - so the check becomes: `method.Parameters[1].Type.Inherits(eventsArgSymbol) || method.Parameters[1].Type.Inherits(routedEventsArgSymbol)`. If we end up identifying more such special event args types, then we can work on a more generic solution.

**Comment by Evangelink** [comment]
Well [WindowCreatedEventArgs](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.windowcreatedeventargs) or [BindingFailedEventArgs](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.bindingfailedeventargs) could be common too.

**Comment by mavasani** [comment]
Ah, unfortunate that these types don't derive from `EventArgs` though. I think your suggestion about suffix might be appropriate then: `method.Parameters[1].Type.Inherits(eventsArgSymbol) || method.Parameters[1].Type.Name.EndsWith("EventArgs")`
